code/game/machinery/doors/airlock.dm

```diff
@@ -299,6 +299,7 @@
 	if(locked)
 		return
 	locked = TRUE
+	SEND_SIGNAL(src, COMSIG_AIRLOCK_BOLT)
```

Comment:
Can you put this signal on a setter for `locked` instead?
Something like this:
```dm
set_locked(new_value)
	if(locked == new_value)
		return
	SEND_SIGNAL(src, COMSIG_AIRLOCK_SET_LOCKED, new_value)
	. = locked
	locked = new_value
```
This accomplishes a few things:
* The event reports a change in state, which is more extensible. If we had three states we'd need three signals otherwise.
* It lets more things react to the change in this value, as they can redefine the proc and check whether the parent return is `null`, to know whether there was a change or not.
* It informs things before the change happens, so it allows for us to easily add something that blocks the change without needing to refactor further existing logic.